lib/orm.php

```diff
@@ -2205,9 +2205,15 @@ public function save() {
 	 *
 	 * @example From the Indexable_Link_Builder class: $this->seo_links_repository->query()->save_many( $links );
 	 *
+	 * @throws \Exception Invalid or no instances to be inserted.
 	 * @throws \Exception Instance to be inserted is not a new one.
 	 */
 	public function insert_many( $models ) {
+		// Validate the input first.
+		if ( ! \is_array( $models ) || empty( $models ) ) {
+			throw new \Exception( 'Invalid or no instances to be inserted' );
```

Comment:
```suggestion
			throw new \InvalidArgumentException( 'Invalid or no instances to be inserted' );
```
[`InvalidArgumentException`](https://www.php.net/manual/en/class.invalidargumentexception.php) is more specific and would make exception handling a bit easier. Since we can more easily determine the cause of the exception (an invalid argument to a method, in this case).

I would also do the same for the other exception thrown in this code: change it from a general `Exception` to an `InvalidArgumentException`.